### ISSUE TITLE
Add our own coverage decoration

### ISSUE THREAD

**Issue by davidanthoff** [julia-vscode/julia-vscode#990]
Right now we can create lcov files, but it is brittle. I think we should instead just create our own solution, with text decorations it doesn't look that difficult to me. We could just directly read the .cov files that Julia creates, then we also wouldn't need to do any conversions.

**Comment by vchuravy** [comment]
Given that Julia supports lcov directly (https://github.com/JuliaLang/julia/pull/30381) might be easier to just add a way to easily produce those?

**Comment by davidanthoff** [comment]
Yeah, not clear to me what file format is best here... Maybe we should even skip storing anything on disc entirely?

**Comment by ZacLN** [comment]
I'm not completely clear on what's being discussed/proposed here (or the distintion between lcov and cov). Is this just to use .cov files to highlight the paired .jl file?

**Comment by davidanthoff** [comment]
Yes :) We already support running tests with coverage, and then a lcov file written into the root of the workspace. One can then use a different extension (like coverage gutters) to show coverage results in the editor, but it requires another extension that is a bit brittle etc. Seems we could do better by just having this out-of-the box.

**Comment by tk3369** [comment]
I concur - it would be best doing this ourselves than relying on another extension.

BTW, I personally found the .cov files pretty annoying where they clutter up the file browser.  Looks like LCOV is a better format and it's a single file?  That sounds like to me 😄 

I think tools like codecov is pretty nice but it always feel clunky for having to commit and push to GitHub before getting a coverage report.  I believe integrating with VSCode is the best solution that improves developer productivity and quality.
